server/config/lib/app.js

```diff
@@ -6,32 +6,54 @@
 var config = require('../config'),
   mongoose = require('./mongoose'),
   express = require('./express'),
-  chalk = require('chalk'),
-  seed = require('./seed');
-
-function seedDB() {
-  if (config.seedDB && config.seedDB.seed) {
-    console.log(chalk.bold.red('Warning:  Database seeding is turned on'));
-    seed.start();
-  }
-}
-
-// Initialize Models
-mongoose.loadModels(seedDB);
+  chalk = require('chalk');
 
-module.exports.init = function init(callback) {
-  mongoose.connect(function (db) {
-    // Initialize express
-    var app = express.init(db);
-    if (callback) callback(app, db, config);
+// Establish an SQL server connection, instantiating all models and schemas
+function startSequelize() {
+  return new Promise(function (resolve, reject) {
+    let orm = null;
+    if (config.orm) {
+      orm = require('./sequelize');
+      orm.sync()
+        .then(function () {
+          resolve(orm);
+        });
+    }
+  });
+}
 
+// Establish a MongoDB connection, instantiating all models
+function startMongoose() {
+  return new Promise(function (resolve, reject) {
+    mongoose.loadModels()
+      .then(mongoose.connect)
+      .then(mongoose.seed)
+      .then(function(dbConnection) {
+        resolve(dbConnection);
+      })
   });
-};
+}
+
+/**
+ * Establish ExpressJS powered web server
+ * @param {object} db a Mongoose DB
+ * @param {object} orm an SQL DB
+ */
+function startExpress(db, orm) {
+  return new Promise(function (resolve, reject) {
+    // Initialize the ExpressJS web application server
+    const app = express.init(db, orm);
+    resolve(app);
+  })
+}
 
-module.exports.start = function start(callback) {
-  var _this = this;
 
-  _this.init(function (app, db, config) {
+// Boot up the server
+module.exports.start = function start() {
+  return new Promise(async function (resolve, reject) {
+    const orm = await startSequelize();
```

Comment:
@lirantal Why start both Sequelize & Mongoose by default? I understand there could be cases where it's useful to have both running side-by-side. However, it seems a little unclear as to why it's done like this by default, since the configuration option is also set by default.